azure-iot-device/azure/iot/device/provisioning/pipeline/provisioning_pipeline.py

```diff
@@ -65,11 +71,16 @@ def remove_this_code(call):
             if call.error:
                 raise call.error
 
-        self._pipeline.run_op(
-            pipeline_ops_provisioning.SetSymmetricKeySecurityClient(
+        if isinstance(security_client, X509SecurityClient):
+            op = pipeline_ops_provisioning.SetX509SecurityClient(
                 security_client=security_client, callback=remove_this_code
             )
-        )
+        else:
```

Comment:
>else: [](start = 8, length = 5)

Probably should be an elif, with else raising some error just to be safe

Reply:
i am not completely sure of the error yet. Previously when there was only 1 security client it was just doing that. now that we have 2 i am checking for x509 otherwise going to Symmetric. I have given a log message now but will need to think about raising

---
In reply to: [291749913](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291749913) [](ancestors = 291749913)

Reply:
Could probably do a ValueError or something for the time being.

---
In reply to: [291789465](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291789465) [](ancestors = 291789465,291749913)